src/dodal/log.py

```diff
@@ -21,6 +21,49 @@
 )
 
 
+class EnhancedRollingFileHandler(logging.handlers.TimedRotatingFileHandler):
+    """Combines features of TimedRotatingFileHandler and RotatingFileHandler"""
+
+    def __init__(
+        self,
+        filename,
+        when="D",
+        interval=1,
+        backupCount=10,
+        encoding=None,
+        delay=0,
+        utc=0,
+        maxBytes=1e8,
+    ):
+        logging.handlers.TimedRotatingFileHandler.__init__(
+            self, filename, when, interval, backupCount, encoding, delay, utc
+        )
+        self.maxBytes = maxBytes
+
+    def shouldRollover(self, record):
+        """
+        Check file size and times to see if rollover should occur
+        """
+        time = super().shouldRollover(record)
+        if time:
+            return 1
+        else:
+            if self.stream is None:  # delay was set...
+                self.stream = self._open()
+            if self.maxBytes > 0:  # are we rolling over?
+                msg = "%s\n" % self.format(record)
+                self.stream.seek(0, 2)  # due to non-posix-compliant Windows feature
+                if self.stream.tell() + len(msg) >= self.maxBytes:
+                    return 1
+        return 0
```

Comment:
We can simplify this a bit by doing our check first then just returning the result of the super check